### ISSUE TITLE
Modkit Extract Kmer Size Limit

### ISSUE THREAD

**Issue by andrewgalbraith21** [nanoporetech/modkit#121]
Hello! 

I was wondering why a size limit of 12 was placed on the kmer_size for modkit extract? For my specific utility it would be nice to have this expanded to 17. I could likely alter the code myself to make this change but just was wondering why the size limit was put in place. Thanks!

All the best,
Andrew Galbraith

**Comment by ArtRand** [comment]
Hello @andrewgalbraith21,

The limit is mostly an implementation detail/performance tweak. I can get you a build tomorrow with an increased limit. I'll also put it on the roadmap to investigate removing the limit all together.

**Comment by andrewgalbraith21** [comment]
Hello @ArtRand,

Thanks so much I appreciate the timeliness!

**Comment by ArtRand** [comment]
@andrewgalbraith21 If I push the changes necessary to a branch, are you able to build the project and test it?

**Comment by andrewgalbraith21** [comment]
Hello @ArtRand, I could try although last time I tried build to modkit using cargo it threw an error so I've only been using the precompiled binaries.

**Comment by ArtRand** [comment]
@andrewgalbraith21, gotcha, let me try and get you a compiled ready-to-go build then.

**Comment by ArtRand** [comment]
Hello @andrewgalbraith21, sorry about the delay. I've attached a build with a larger max kmer size. I just need to do some testing before I can say whether or not this implementation will be in the mainline version. But this should unblock you in the mean time. 
[modkit_v0.2.5-dev3180523_centos7_x86_64.tar.gz](https://github.com/nanoporetech/modkit/files/14131748/modkit_v0.2.5-dev3180523_centos7_x86_64.tar.gz)

**Comment by andrewgalbraith21** [comment]
Hello @ArtRand, sorry for the delay and thanks for the build. It appears to be working fine for me!

**Comment by ArtRand** [comment]
@andrewgalbraith21 This feature was added to [v0.2.5](https://github.com/nanoporetech/modkit/releases/tag/v0.2.5).

**Comment by andrewgalbraith21** [comment]
@ArtRand perfect thanks so much! This tool is very helpful
